packages/redux/src/products/actions/factories/fetchProductGroupingFactory.ts

```diff
@@ -37,19 +40,29 @@ const fetchProductGroupingFactory =
     query: GroupingQuery = {},
     config?: Config,
   ) =>
-  async (dispatch: Dispatch): Promise<ProductGrouping> => {
+  async (
+    dispatch: Dispatch,
+    getState: () => StoreState,
+  ): Promise<ProductGrouping> => {
     try {
       dispatch({
         meta: { productId },
         type: FETCH_PRODUCT_GROUPING_REQUEST,
       });
 
       const result = await getProductGrouping(productId, query, config);
+      const state = getState();
+      const product = getProduct(state, productId);
+      const previousGroupings = product?.grouping;
+      const queryString = query && buildQueryStringFromObject(query);
+      const hash = queryString ? queryString : '!all';
 
-      // Replace all the grouping data with the new one
       const productWithGrouping = {
         id: productId,
-        grouping: result,
+        grouping: {
```

Comment:
Instead of doing this, what do you think of using an entity mapper for merging the results in the store? If you use an entity mapper for the `FETCH_PRODUCT_GROUPING_SUCCESS` action, you can perform the logic to merge all the groupings instead of having the action factory retrieve the previous groupings from the state and merge them.